kubernetes/patroni_k8s.yaml

```diff
@@ -120,3 +121,60 @@ type: Opaque
 data:
   superuser-password: emFsYW5kbw==
   replication-password: cmVwLXBhc3M=
+
+---
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: patronidemo
+
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: Role
+metadata:
+  name: patronidemo
+rules:
+- apiGroups:
+  - ""
+  resources:
+  - configmaps
+  verbs:
+  - create
+  - get
+  - list
+  - patch
+  - update
+  - watch
+- apiGroups:
+  - ""
+  resourceNames:
+  - patronidemo
+  resources:
+  - endpoints
+  verbs:
```

Comment:
Patroni also needs to be able to `list` and `watch` Endpoints if `kubernetes.use_endpoint: true`
Also in this case it doesn't need have an access to `ConfigMaps`.

Reply:
Hmmm... The options are:
1. document both approaches with roles specific for each use case
2. create uber role with all in
wdyt?

Reply:
For the demo example I think I'd go with the latter with comment that this is for configmap approach and the other is for endpoints.

Reply:
why not do the grants for both things in the default roles/template, and document how to restrict them based on which mode you're running in?

Reply:
Yeah, I'll update it later today. That's I guess the best approach.